powershell/public/maester/exchange/Test-MtLimitOnMicrosoftDomainUsage.ps1

```diff
@@ -0,0 +1,43 @@
+﻿<#
+.SYNOPSIS
+    Ensure mailboxes do not use the .onmicrosoft.com domain as primary SMTP address
+
+.DESCRIPTION
+    This test checks if any mailbox is using the .onmicrosoft.com domain as primary SMTP address.
+    Usage of the .onmicrosoft.com domain has its limitation and receives throttling.
+    
+.EXAMPLE
+    Test-MtLimitOnMicrosoftDomainUsage
+
+    Returns true if no mailbox is using the .onmicrosoft.com domain as primary SMTP address
+
+.LINK
+    https://maester.dev/docs/commands/Test-MtLimitOnMicrosoftDomainUsage
+#>
+function Test-MtLimitOnMicrosoftDomainUsage {
+    [CmdletBinding()]
+    [OutputType([bool])]
+    param()
+    
+    if (-not (Test-MtConnection ExchangeOnline)) {
+        Add-MtTestResultDetail -SkippedBecause NotConnectedExchange
+        return $null
+    } 
+    $return = $true
+    Write-Verbose "Checking if mailboxes use the .onmicrosoft.com domain as primary SMTP address..."
+
```

Comment:
Please remember to wrap the test in a try/catch block so any errors can be properly reported in the results. Thank you! See [the docs](https://maester.dev/docs/writing-tests/advanced-concepts#step-2-create-test-functions-file) for more details.